src/sd/spi.rs

```diff
@@ -0,0 +1,179 @@
+use crate::sys::*;
+
+use core::ops::Deref;
+
+use esp_idf_hal::{gpio, peripheral, spi};
+
+pub struct SpiDeviceBuilder(sdspi_device_config_t);
+
+impl From<sdspi_device_config_t> for SpiDeviceBuilder {
+    fn from(config: sdspi_device_config_t) -> Self {
+        Self(config)
+    }
+}
+
+impl Default for SpiDeviceBuilder {
+    fn default() -> Self {
+        Self(sdspi_device_config_t {
+            host_id: spi_host_device_t_SPI2_HOST,
+            gpio_cs: gpio_num_t_GPIO_NUM_13,
+            gpio_cd: gpio_num_t_GPIO_NUM_NC,
+            gpio_wp: gpio_num_t_GPIO_NUM_NC,
+            gpio_int: gpio_num_t_GPIO_NUM_NC,
+            #[cfg(esp_idf_version = "5.2")]
```

Comment:
this would only work for a specific 5.2 version. So a 5.3 and onwards would break it. 

One way is to negate the logic and say every version that is not 4 & 5.0 & 5.1 will work, that way its future proof. 

```rust
 #[cfg(not(any(
    esp_idf_version_major = "4",
    all(esp_idf_version_major = "5", esp_idf_version_minor = "0"),
    all(esp_idf_version_major = "5", esp_idf_version_minor = "1"),
 )))]
 
 ```
 
Since we only can use any(), all(), or not() and no greater or less that's the best we got ;D

Reply:
Alright, it is somewhat bulky indeed, but thank you nonetheless ^^.